frame/staking-rewards/src/lib.rs

```diff
@@ -378,13 +423,29 @@ pub mod pallet {
 		}
 
 		/// Extends fNFT position stake. Applied only to next epoch.
+		///
+		/// Is better then unstake and restake more in case of penalty.
+		/// But not incentivises to making many small stakes one on other to game over system.
 		#[pallet::weight(10_000)]
+		#[transactional]
 		pub fn extend_stake(
-			_origin: OriginFor<T>,
-			_instance_id: InstanceIdOf<T>,
-			_balance: T::Balance,
+			origin: OriginFor<T>,
+			instance_id: InstanceIdOf<T>,
+			balance: T::Balance,
+			keep_alive: bool,
 		) -> DispatchResult {
-			Err(DispatchError::Other("no implemented. TODO: insert update for next fold"))
+			// TODO: consider attack adding near zero stake each block, preventing stream to finish
+			let owner = ensure_signed(origin)?;
+			T::ensure_protocol_nft_owner::<StakingNFTOf<T>>(&owner, &instance_id)?;
+			let position = T::get_protocol_nft::<StakingNFTOf<T>>(&instance_id)?;
+			let protocol_account = Self::account_id(&position.asset);
+			T::Assets::transfer(position.asset, &owner, &protocol_account, balance, keep_alive)?;
```

Comment:
I wonder if it's a good idea to transfer to the protocol account(unless a sub account) as protocol account also stores rewards, there are cases where staked asset is same as rewarded asset

Reply:
i will improve this next prs. there will be protocol and asset, and locked compounding vs not.